app/models/agents/webhook_agent.rb

```diff
@@ -84,7 +92,8 @@ def receive_web_request(params, method, format)
         JSON.parse(response.body)['success'] or
           return ["Not Authorized", 401]
       end
-
+      params[interpolated['header_key']] = header_for(headers)
```

Comment:
I think we should check if `interpolated['header_key']` is present and only apply the headers in that case. Otherwise we would write them as `nil` in `params`.

Reply:
Ok. I actually moved it around a bit. We wont add the headers if they header_key is empty, but I've also moved the logic inside the payload_for loop so the headers are present regardless of payload_path.